embassy-mcxa/src/wwdt.rs

```diff
@@ -0,0 +1,220 @@
+        unsafe {
+            crate::pac::Interrupt::WWDT0.enable();
+        }
+        Self {
+            _peri,
+            load_value: 0,
+            frequency: _frequency / 4,
+            info: unsafe { &*pac::Wwdt0::ptr() },
+        }
+    }
+
+    // Start the watchdog timer with the specified timeout period.
+    ///
+    /// # Arguments
+    ///
+    /// * `period` - The timeout period after which the watchdog will trigger
+    /// * `warning` - Optional warning interrupt period before timeout. If `Some`, an interrupt
+    ///               will be generated at this time before the actual timeout. If `None`, the
+    ///               watchdog will directly reset the system on timeout.
+    ///
+    pub fn start(&mut self, period: Duration, warning: Option<Duration>) {
+        let timeout_cycles = (self.frequency as u64 * period.as_micros()) / 1_000_000;
+
+        // Ensure the value fits in u32 and is within valid range
+        // Writing a value below FFh causes 00_00FFh to load into the register. Therefore, the minimum timeout interval is TWDCLK X 256 X 4.
+        assert!(timeout_cycles <= 0xFFFFFF, "Timeout too large for watchdog counter");
+        assert!(timeout_cycles > 0xFF, "Timeout too small for watchdog counter");
+
+        self.load_value = timeout_cycles as u32;
+        self.set_timeout_value(self.load_value);
+
+        //Windows value is set to max at reset for no effect.
+
+        if let Some(warning_value) = warning {
+            let warning_cycles = (self.frequency as u64 * warning_value.as_micros()) / 1_000_000;
+            assert!(
+                warning_cycles <= 0x3FF,
+                "Warning value too large. At 1MHz, warning value must be <= 4.092ms."
+            );
+
+            self.set_warning_value(warning_cycles as u16);
+            self.enable_interrupt();
+        } else {
+            self.enable_reset();
+        }
+
+        self.lock_oscillator();
+        self.enable(true);
+        self.feed();
+
+        // Set the WDPROTECT bit to false after the Feed Sequence (0xAA, 0x55)
+        self.set_protect_mode(false);
+    }
+
+    /// Feed the watchdog to prevent timeout.
+    ///
+    /// This must be called periodically before the timeout period expires to prevent
+    /// the watchdog from triggering a reset or interrupt.
+    pub fn feed(&self) {
+        //TBD To avoid an unintended interrupt, it is a good practice to disable interrupts around a feed sequence
+        self.info.feed().write(|w| unsafe { w.feed().bits(0xAA) });
+        self.info.feed().write(|w| unsafe { w.feed().bits(0x55) });
+    }
+
+    /// Enable or disable the watchdog timer.
+    ///
+    /// # Arguments
+    ///
+    /// * `bit` - `true` to enable (run), `false` to disable (stop).
+    fn enable(&self, bit: bool) {
+        self.info
+            .mod_()
+            .modify(|_, w| if bit { w.wden().run() } else { w.wden().stop() });
+        while self.info.tc().read().count() == 0xFF {}
+    }
```

Comment:
I'm not sure I like this raw boolean. Maybe define an enum or split into `enable()` and `disable()` APIs?

Additionally, you can get rid of the `if` by using the `bit()` method:

```rust
.modify(|_, w| w.wden().bit(bit));
```